### ISSUE TITLE
The new text for SC2.4.11: Focus Visible (Enhanced)

### ISSUE THREAD

**Issue by guyhickling** [w3c/wcag#1115]
I am very happy that revised texts have now been considered for the Focus Visible (Enhanced) SC, as posted at [https://docs.google.com/document/d/1F83m-HXRkXz1QCF6_QNtQGzIULugMiFSLgtqDRgKA-s/edit#](https://docs.google.com/document/d/1F83m-HXRkXz1QCF6_QNtQGzIULugMiFSLgtqDRgKA-s/edit#). They are a huge improvement. 

Looking at those options, I would advocate a combination of options 1 and 2. I am greatly in favor of the 2px minimum thickness of the line, so would like that from Option 1. For the rest of the SC I like the Option 2 version. It states things clearly, and the 3 to 1 contrast with adjacent colors (which option 1 doesn't have) is definitely necessary.

I strongly favor the 2px line thickness because, for keyboard users navigating using the Tab key, if they have poor vision, it is so difficult to spot where the focus has just just jumped to as they Tab around the page. The more complex and busy the page, the more difficult it becomes. (Even when just testing web pages, with my below average vision, I am often slow to see where it's gone!) Making it 2px in thickness makes it so much easier for these people - and lots of websites from larger organizations and businesses use an outline of 3 or even 4px, so it is a well accepted method. Yes it's prescriptive, but focus outlines are only seen at length by keyboard users - who are the very ones who need a good one. (Other people see them only briefly while a new page loads, and even that can be stopped by a bit of scripting.) 

I hesitate to disturb things any more beyond that in case it leads us too far away from these new options, but regarding the matter of colour-change indicators, where the fore and background colors both change to indicate focus, I think that could still be included with only a few words extra. How about adding a clause to the first bullet point to make it:

> "Is at least the size of a 2 CSS pixel solid border around the control, or is a change of fore and background colors of the component, and"

(An extra three or four words would also be needed in the third bullet point to correspond with that.)

Anyway, congratulations on a hugely improved offering, and many thanks to the Group for listening to all our previous public discussions about it.

**Comment by alastc** [comment]
Thanks Guy, we include that feedback in the upcoming discussion.

**Comment by alastc** [comment]
Hi Guy,

This was discussed at the AG meeting on Tuesday, and the general consensus was to keep some flexibility rather than prescribing a particular focus style/approach. 

(In my opinion:) Across all the various scenarios, requiring a 2px (or equivalent) size would make some very visible indicators fail. The current definition has increased from the first draft and will have a significant (and good) impact whilst being flexible across different designs.

You can see the updates in a branch, the [understanding doc](https://raw.githack.com/w3c/wcag/wcag22-focus-visible-enh-updates/understanding/22/focus-visible-enhanced.html) (including SC text) can be previewed.

Kind regards,

Alastair

**Comment by alastc** [comment]
Noting that Bruce suggested adding option 1 as an AAA SC. That might have the effect of making the AA version clearer?

**Comment by alastc** [comment]
Hi @guyhickling,

The [group agreed](https://www.w3.org/2020/06/30-ag-minutes.html#item04) to add a triple-A version which includes the 2px with a 4.5:1 contrast ratio for the change.

I think that is the likely extent of the changes for this issue without new evidence of a previously-unknown scenario, or something else that is new.

Closing this issue, please reopen if you think there is something that has not been considered.
